staging/src/k8s.io/apiserver/pkg/storageversion/manager.go

```diff
@@ -35,6 +36,11 @@ import (
 type ResourceInfo struct {
 	GroupResource schema.GroupResource
 
+	// Optional. Overwrites the group name when creating the StorageVersion object.
+	// Used by the core group to make the group name more readable and a valid DNS
+	// subdomain.
+	OverwriteGroupName string
```

Comment:
if it's only used by the core group, can we hardcode the logic instead?

Reply:
Do you mean having a boolean instead of the string?

Reply:
We can also hardcode the logic before we create the StorageVersion object. A downside of that is if an aggregated server accidentally register a resource info for an aggregated API, but omitted the group. The manager will create a StorageVersion with `group=core` for it

Reply:
If an aggregated server tries to register a `group=""` resource. The storage migrator won't be able to migrate the resource anyway. 

I updated the code to hardcode the logic.